src/SigasiProjectCreator/ConverterHelper.py

```diff
@@ -4,149 +4,457 @@
         if verilog_includes is not None and len(verilog_includes) > 0:
+            verilog_includes = [pathlib.Path(include_path) for include_path in verilog_includes]
             print("Includes: " + str(verilog_includes))
         if verilog_defines is not None and len(verilog_defines) > 0:
             print("Defines: " + str(verilog_defines))
-        linked_files = parser_output.linked_file_mapping
-        if linked_files is not None and len(linked_files) > 0:
-            print("Linked files: " + str(linked_files))
     else:
         entries = parser_output
+
+    new_entries = dict()
+    for path, library in entries.items():
+        new_entries[pathlib.Path(path)] = library
+    entries = new_entries
     print("Library mapping: " + str(entries))
 
+    if not ArgsAndFileParser.get_skip_check_exists():
+        for file in entries.keys():
+            assert file.is_file(), f'*ERROR* file {file} does not exist'
+
     sigasi_project_file_creator = SigasiProjectCreator(project_name)
     sigasi_project_file_creator.unmap("/")
 
-    forceVHDL = False
-    forceVerilog = False
-    forceVUnit = False
-
-    linked_folders = dict()
-    abs_destination = absnormpath(destination)
-    for path, library in entries.items():
-        abs_path = absnormpath(path)
-        relative_path = os.path.relpath(abs_path, abs_destination)
-        if (not forceVerilog) and (relative_path.endswith('.v') or relative_path.endswith('.sv')):
-            forceVerilog = True
-        if (not forceVHDL) and (relative_path.endswith('.vhd') or relative_path.endswith('.vhdl')):
-            forceVHDL = True
-        if not relative_path.startswith(".."):
-            sigasi_project_file_creator.add_mapping(relative_path, library)
-        else:
-            common_prefix = os.path.dirname(os.path.commonpath([p + os.path.sep for p in [abs_path, abs_destination]]))
-            eclipse_path = os.path.relpath(abs_path, common_prefix)
-            directory_name = get_parts(eclipse_path)[-1]
-            if str(ArgsAndFileParser.get_layout_option()) == 'default':
-                target = os.path.join(common_prefix, directory_name)
-            else:
-                target = 'virtual:/virtual'
-
-            linked_folders[directory_name] = target
+    has_vhdl = False
+    has_verilog = False
+    project_layout = ArgsAndFileParser.get_layout_option()
+    if project_layout == 'simulator':
+        (has_vhdl, has_verilog) = create_project_simulator(sigasi_project_file_creator, entries)
+    elif project_layout == 'linked-files-flat':
+        (has_vhdl, has_verilog) = create_project_links_flat(sigasi_project_file_creator, entries)
+    elif project_layout == 'linked-files-tree':
+        (has_vhdl, has_verilog) = create_project_links_tree(sigasi_project_file_creator, entries)
+    elif project_layout == 'linked-folders':
+        (has_vhdl, has_verilog) = create_project_links_folders(sigasi_project_file_creator, entries)
```

Comment:
It'd be nice if these were different subclasses of `SigasiProjectCreator` class, an instance of which could be created in the parser and then used here.

Reply:
Converted to class `ProjectCreator` + subclasses for each project layout